bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -141,9 +152,25 @@ public Builder setStatsLogger(StatsLogger statsLogger) {
             return this;
         }
 
+
+        public Builder dnsResolver(DNSToSwitchMapping dnsResolver) {
+            this.dnsResolver = dnsResolver;
+            return this;
+        }
+
+        public Builder requestTimer(HashedWheelTimer requestTimer) {
```

Comment:
@sijie  I agree that it would be useful but IMHO  as hashedwelltimer is from netty we are going to introduce an hard dependency to that third party library on the top level API.
Maybe it would be better to have a generic  'custom environment' Map<String,Object> to be passed to every subcomponent initialization